app/src/org/commcare/tasks/EntityCacheInvalidationWorker.kt

```diff
@@ -13,10 +14,17 @@ class EntityCacheInvalidationWorker(context: Context, workerParams: WorkerParame
     override suspend fun doWork(): Result {
         try {
             val entityStorageCache: CommCareEntityStorageCache = CommCareEntityStorageCache("case")
-            entityStorageCache.processShallowRecords()
+            if (!entityStorageCache.isEmpty) {
+                entityStorageCache.processShallowRecords()
+            }
         } catch (e: Exception) {
             Logger.exception("Error encountered while invalidating entity cache", e)
             return Result.failure()
+        } finally {
+            // we want to schedule the Prime worker irrespective of result of this task
+            if (CommCareApplication.isSessionActive()) {
+                PrimeEntityCacheHelper.schedulePrimeEntityCacheWorker()
```

Comment:
Honestly, I don't like that these two methods refer to different static singletons with different lifecycles. I'm worried this code has the risk of further spreading out access to static / singleton lifecycles while also being fairly likely to be executed in contexts where those lifecycles are changing (right after login, or when a user has completed a form and is logging out for the day). 

My guess here is that it's better to make "the other half" of any calls like this that depend on multiple singleton contexts get pushed into one of them, so that at least there's no ambiguity at runtime about which memory space or sandbox the second call is going to be running against.

Reply:
Not completely sure if I got you right but did you mean [like this](https://github.com/dimagi/commcare-android/pull/2982/commits/091cd01f1ba904cbf3f5115120061193330666e7) ?